src/Platform/Microsoft.Testing.Platform/Helpers/System/SystemConsole.cs

```diff
@@ -45,12 +53,38 @@ static SystemConsole()
 
     // This is based on the UnsupportedOSPlatformAttribute on the API:
     // https://github.com/dotnet/runtime/blob/8a79637e1454c751c24a984b7fb8af4c4d43394b/src/libraries/System.Console/src/System/Console.cs#L560-L563
+    [SupportedOSPlatformGuard("android")]
+    [SupportedOSPlatformGuard("ios")]
+    [SupportedOSPlatformGuard("tvos")]
+    [SupportedOSPlatformGuard("browser")]
     private static bool IsCancelKeyPressNotSupported()
         => RuntimeInformation.IsOSPlatform(OSPlatform.Create("ANDROID")) ||
             RuntimeInformation.IsOSPlatform(OSPlatform.Create("IOS")) ||
             RuntimeInformation.IsOSPlatform(OSPlatform.Create("TVOS")) ||
             RuntimeInformation.IsOSPlatform(OSPlatform.Create("BROWSER"));
 
+    // This is based on the UnsupportedOSPlatformAttribute on the API:
+    // https://github.com/dotnet/runtime/blob/d6cfa2039309c1d5cef630e95eb5274a556dc726/src/libraries/System.Console/src/System/Console.cs#L334-L337
+    [SupportedOSPlatformGuard("android")]
+    [SupportedOSPlatformGuard("ios")]
+    [SupportedOSPlatformGuard("tvos")]
+    [SupportedOSPlatformGuard("browser")]
+    private static bool IsForegroundColorNotSupported()
```

Comment:
Can we drop the 3 `Is...` and instead just annotate the API?

Reply:
The other APIs (CancelKeyPress, ForegroundColor, Clear), we want to make them callable on any platform. So they are *supported*, but we make them no-op on those platforms.

Reply:
Don't we instead want to push the decision of calling or not to the caller? Like it's done on Console.

Reply:
@Evangelink The *existing* code was already to try to be no-op on Android and iOS.

Generally, I think whether or not to push the decision to the caller depends on the API itself. If it's really an important functionality, push to caller

Reply:
Addressed per our discussion offline.